Add MFA support

kevdliu opened the pull request:
First attempt at adding MFA support. Overview of changes:
-  Added --mfa argument to the CLI which allows the user to provide a MFA code for login.
- Since Abode uniquely identifies clients by a combination of the UUID and a session cookie, I modified the cache to persist session cookies to disk after successfully logging in and loading it back into requests.session during initialization. 
- Added some constants and error for MFA
- I had to modify the CLI logic a little bit for when a MFA code is provided. I decided against adding mfa_code as an argument and a property to the Abode class since the code is very time sensitive and there's no guarantee that it'll be used for login immediately in that fashion. If the user provides a MFA code, we'll log in using the login() function first before fetching devices from Abode.

MisterWil commented on abodepy/__init__.py:
```diff
@@ -137,6 +148,14 @@ def login(self, username=None, password=None):
 
         response_object = json.loads(response.text)
 
+        if response_object.get('mfa_type', None) == "google_authenticator":
```
The only thing I might consider is adding an else to this block where if `mfa_type` exists in the response with any other value, return another exception of a type like `ERROR.UNKNOWN_MFA_REQUIRED`. That way if Abode ever does add another MFA type we can handle it gracefully.

kevdliu replied:
good call, ill add an error for unknown mfa type

MisterWil commented:
This looks great! Thank you so much for working on this. I really appreciate the tests too. I left a single comment, and really it's only a suggestion. If you feel like it's working well feel free to mark it ready for review and I'll merge it in and push out a new version.

kevdliu commented:
Good call on handling unknown mfa types. I changed the PR to include that and added some related tests. I've been using this for the past week and no problems so far 👍

MisterWil commented:
Great! Thank you!
